attentive-android-sdk/src/main/java/com/attentive/androidsdk/AttentiveApi.java

```diff
@@ -0,0 +1,187 @@
+        });
+    }
+
+    private void internalSendUserIdentifiersCollectedEventAsync(String geoAdjustedDomain, UserIdentifiers userIdentifiers, AttentiveApiCallback callback) {
+        String externalVendorIdsJson = null;
+        try {
+            List<ExternalVendorId> externalVendorIds = buildExternalVendorIds(userIdentifiers);
+            externalVendorIdsJson = objectMapper.writeValueAsString(externalVendorIds);
+        } catch (JsonProcessingException e) {
+            callback.onFailure(String.format("Could not serialize the UserIdentifiers. Message: '%s'", e.getMessage()));
+            return;
+        }
+
+        String metadataJson = null;
+        try {
+            Metadata metadata = buildMetadata(userIdentifiers);
+            metadataJson = objectMapper.writeValueAsString(metadata);
+        } catch (JsonProcessingException e) {
+            callback.onFailure(String.format("Could not serialize metadata. Message: '%s'", e.getMessage()));
+            return;
+        }
+
+        HttpUrl url = getHttpUrlEventsEndpointBuilder()
+                .addQueryParameter("v", "4.16.10_f0149441bf")
+                .addQueryParameter("c", geoAdjustedDomain)
+                .addQueryParameter("t", "idn")
+                .addQueryParameter("evs", externalVendorIdsJson)
+                .addQueryParameter("m", metadataJson)
+                .build();
+
+        Request request = new Request.Builder().url(url).build();
+        httpClient.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(@NonNull Call call, @NonNull IOException e) {
+                callback.onFailure(String.format("Error when calling the event endpoint: '%s'", e.getMessage()));
+            }
+
+            @Override
+            public void onResponse(@NonNull Call call, @NonNull Response response) {
+                if (!response.isSuccessful()) {
+                    callback.onFailure(String.format("Invalid response code when calling the event endpoint: '%d', message: '%s'", response.code(), response.message()));
+                    return;
+                }
+
+                callback.onSuccess();
+            }
+        });
+    }
+
+    private HttpUrl.Builder getHttpUrlEventsEndpointBuilder() {
+        return new HttpUrl.Builder()
+                .scheme("https")
+                .host(ATTENTIVE_EVENTS_ENDPOINT_HOST)
+                .addPathSegment("e");
+    }
+
+    @Nullable
+    private String parseAttentiveDomainFromTag(String tag) {
+        Pattern pattern = Pattern.compile("window.__attentive_domain='(.*?).attn.tv'");
+        Matcher matcher = pattern.matcher(tag);
+        if (matcher.find()) {
+            if (matcher.groupCount() == 1) {
+                return matcher.group(1);
+            }
+        }
+
+        return null;
+    }
+
+    private Metadata buildMetadata(UserIdentifiers userIdentifiers) {
+        Metadata metadata = new Metadata();
+        metadata.setSource("msdk");
```

Comment:
nit: do we need this if msdk is already the default source?

Reply:
nope! deleted. thank you